corehq/form_processor/models/forms.py

```diff
@@ -12,15 +12,32 @@
 from dimagi.utils.couch.undo import DELETED_SUFFIX
 
 from corehq.blobs.exceptions import NotFound
-from corehq.sql_db.models import PartitionedModel
+from corehq.sql_db.models import PartitionedModel, RequireDBManager
 
-from ..exceptions import AttachmentNotFound, MissingFormXml
+from ..exceptions import AttachmentNotFound, MissingFormXml, XFormNotFound
 from ..track_related import TrackRelatedChanges
 from .abstract import AbstractXFormInstance
 from .attachment import AttachmentMixin
 from .mixin import SaveStateMixin
 
 
+class XFormInstanceManager(RequireDBManager):
+
+    def get_form(self, form_id, domain=None):
+        """Get form in domain
+
+        This will get a form from any domain if the domain is not provided,
+        but that is deprecated and should be avoided.
+        """
+        try:
+            kwargs = {'form_id': form_id}
+            if domain is not None:
+                kwargs['domain'] = domain
+            return self.partitioned_get(form_id, **kwargs)
```

Comment:
has it been possible to have
1. domain less forms (system forms?)
2. cross domain forms (forms for domain A updating case in domain B). I believe this is possible with case reassignment which would it possible to have a form in case transactions that won't belong to current case domain.
Did we consider that while enforcing domain now?

Reply:
@mkangia 
1. Not sure what you mean. Nothing has changed with regard to system forms or the ability to get a form without specifying the domain.
2. I think that should never happen except in very specific circumstances like possibly in the context of data registries (although I'm not sure if that part of data registries has been implemented yet).

Reply:
Just to clarify (2):
* Forms and cases ALWAYS belong to a single domain.
* All cases updated by a form MUST belong to the same domain as the form

Data registry uses data forwarding to create / update cases in other domains by submitting forms to that domain. See `corehq.motech.repeaters.models.DataRegistryCaseUpdateRepeater`

Reply:
Thanks both.